.github/workflows/main.yml

```diff
@@ -16,19 +16,38 @@ env:
 jobs:
   build:
     runs-on: ubuntu-16.04
-    timeout-minutes: 12
+    timeout-minutes: 60
     steps:
       - uses: actions/checkout@v2
 
       # TODO: cache all the setup
+      - name: Cache
+        uses: actions/cache@v2.1.6
+        id: cache-packages
+        with:
+          path: |
+            /usr/local/bin/protoc
+            /usr/local/bin/protoc-c
+            /usr/local/bin/protoc-gen-c
+            /usr/local/include/google/protobuf
+            /usr/local/include/google/protobuf-c
+            /usr/local/include/protobuf-c
+            /usr/local/lib
```

Comment:
Try and rerun the CI jobs to see if the caching works; if it does, great, but if it doesn't, I worry that caching all of /usr/local/lib will cause issues. I assume there's a particular protobuf library in there that you want to cache, so to cache only that library I'd recommend copying it from /usr/local/lib to some path you can restore (like make ~/.cache/protobuf or something and copy it there) in the install-protobuf step, then copying it back in a step that runs if the cache successfully restored.